[Ref Guide] - Theme variants changes (#731)
dojo/framework: docs/en/styling/supplemental.md

Dojo allows use of modern CSS features such as [custom properties and `var()`](https://www.w3.org/TR/css-variables/) to help abstract and centralize common styling properties within an application.

Rather than having to specify the same values for colors or fonts in every widget's CSS module, abstract custom properties can instead be referenced by name, with values then provided as a theme `variant` within a `.root` class. This separation allows for much simpler maintenance of common styling concerns across an entire application and for theme variants to be created by changing variables.

> Note: You should not import the theme variant file into your widget csss module, it is handled at run time via the `theme.variant()` class.

Reviewer: ```suggestion
Note: do not import the theme variant file into a widget's css module; this is handled instead at run time via the `theme.variant()` class.
```

    -   If specified, [the provided theme](/learn/styling/working-with-themes#writing-a-theme) will act as an override for any theme that the widget may use, and will take precedence over [the application's default theme](/learn/styling/theming-a-dojo-application#making-themeable-applications) as well as [any other theme changes made in the application](/learn/styling/theming-a-dojo-application#changing-the-currently-active-theme).
-   `classes` (optional)
    -   described in the [Passing extra classes to widgets](/learn/styling/theming-a-dojo-application#passing-extra-classes-to-widgets) section.
-   `variant` (optional)
    -   returns the `root` class from the current theme variant.

>     -   should be applied to the root of your widget

Reviewer: ```suggestion
    -   should be applied to the widget's root
```
